### PR TITLE
WASM: Add PlatformNotSupportedException for System.IO.Pipes

### PR DESCRIPTION
**Pull request by akoeplinger**
The library isn't supported on WebAssembly.

### CODE REVIEW

File: src/libraries/System.IO.Pipes/src/System.IO.Pipes.csproj [hunk 1]
```diff
@@ -3,11 +3,14 @@
     <AssemblyName>System.IO.Pipes</AssemblyName>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <OmitTransitiveCompileReferences>true</OmitTransitiveCompileReferences>
-    <TargetFrameworks>$(NetCoreAppCurrent)-Windows_NT;$(NetCoreAppCurrent)-Unix;$(NetCoreAppCurrent)-Browser</TargetFrameworks>
+    <TargetFrameworks>$(NetCoreAppCurrent)-Windows_NT;$(NetCoreAppCurrent)-Unix;$(NetCoreAppCurrent)</TargetFrameworks>
```
Comment (safern) [inline src/libraries/System.IO.Pipes/src/System.IO.Pipes.csproj:6]: Why compile it with `AnyOS` instead? I would like to be intentional of shipping not supported on the platforms it is not supported.

Comment (akoeplinger) [reply src/libraries/System.IO.Pipes/src/System.IO.Pipes.csproj:6]: I think this was the convention we landed on in https://github.com/dotnet/runtime/pull/37944, so any future platform that's not Unix/Windows would also get the PNSE behavior by default.

/cc @Anipik

Comment (safern) [reply src/libraries/System.IO.Pipes/src/System.IO.Pipes.csproj:6]: Yeah this should be fine. I just looked at: https://github.com/dotnet/runtime/blob/master/eng/targetframeworksuffix.props 

and the only platform that will produce a PNSE is `Browser` so either way is fine.

File: src/libraries/System.IO.Pipes/tests/System.IO.Pipes.Tests.csproj [hunk 2]
```diff
@@ -3,6 +3,7 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <IncludeRemoteExecutor>true</IncludeRemoteExecutor>
     <TargetFrameworks>$(NetCoreAppCurrent)-Windows_NT;$(NetCoreAppCurrent)-Unix;$(NetCoreAppCurrent)-Browser</TargetFrameworks>
```
Comment (safern) [inline src/libraries/System.IO.Pipes/tests/System.IO.Pipes.Tests.csproj:5]: Test tfms should match src.

**Review by safern — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]

**Pull request merged**